todos/src/redux/todoSlice.jsx

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export const todoSlice = createSlice({
+  name: "todos",
+  initialState: [],
+  reducers: {
+    addTask: (state, action) => {
+      state.push({
+        id: state.length + 1,
+        todo: action.payload,
+        hasEnded: false,
+      });
+    },
+    endTask: (state, action) => {
+      const todo = state[action.payload];
+      if (todo) {
+        todo.hasEnded = !todo.hasEnded;
+      }
```

Comment:
I left a comment in the previous review I a not sure if you noticed it but here is the quote again
> According to the implementation on this line `todo.hasEnded = !todo.hasEnded;` this reducer will be used to end a todo when it's not ended but will also be used to unEnd it when it's already ended. Naming it endTask only mentions one use-case. It would be better to give a name that reflect both usages.

It's not big therefore I am making it optional for now